src/index.js

```diff
@@ -45,7 +41,50 @@ server.on('after', (req, res, route, err) => {
   }
 });
 
-const startAPI = async () => {
+let qtumProcess;
+
+function startQtumProcess(reindex) {
+  // avoid using path.join for pkg to pack qtumd
+  const qtumdPath = `${path.dirname(__dirname)}/qtumd`;
+  const flags = ['-testnet', '-logevents', '-rpcuser=bodhi', '-rpcpassword=bodhi'];
+  if (reindex) {
+    flags.push('-reindex');
+  }
+
+  qtumProcess = spawn(qtumdPath, flags);
+  logger.debug(`qtumd started on PID ${qtumProcess.pid}`);
+
+  qtumProcess.stdout.on('data', (data) => {
+    logger.debug(`qtumd output: ${data}`);
+  });
+
+  qtumProcess.stderr.on('data', (data) => {
+    logger.error(`qtumd failed with error: ${data}`);
+
+    if (data.includes('You need to rebuild the database using -reindex-chainstate to enable -logevents.')) {
+      // Clean old process first
+      qtumProcess.kill();
+      qtumProcess = null;
```

Comment:
do we need this?

Reply:
I wanted to make sure the first instance is cleaned up just in case the user force closes the app. Not sure if there would be a race condition for `qtumProcess.kill()`. Was thinking there could be a possibility that this call to `kill()` is done before the `SIGINT`'s `kill()` is called.

Not really sure what happens if you call `kill()` twice.

Reply:
@dwalintukan kill() send a signal to the process. After moving qtumprocess var inside the func, we can restart without problem